enhancements/network/egressip_for_ovn_kubernetes.md

```diff
@@ -0,0 +1,134 @@
+
+### Goals & Requirements
+
+2.1. The egress IP functionality for OVN-Kubernetes needs to have feature parity with openshift-sdn, according to 1.1. and 1.2.
+2.2. The egress IP functionality for OVN-Kubernetes should allow pod selection within a namespace when defining the IP of egress traffic. 
+2.3. The egress IP functionality for OVN-Kubernetes should support single-stack IPv4/IPv6 as well as dual-stack, in line with global OpenShift requirements for all networking functionality for release 4.6.
+
+This proposal details how OVN-Kubernetes goes about addressing these goals in the section [Proposal](## Proposal)
+ 
+### Constraints
+
+3.1. No pod can match multiple `EgressIP` objects, leading to multiple egress IP assignments at the same time. 
+3.2. Any egress IP can only be assigned to one node, and one node only, at any given moment in time. 
+3.3. An `EgressIP` object can not claim multiple egress IPs for the same node. 
+
+This proposal details how OVN-Kubernetes goes about addressing these constraints in the section [Proposal](## Proposal)
+ 
+### Non-Goals
+
+4.1. Making an EgressIP object cluster-wide, it will still be namespaced
+
+## Proposal
+
+We propose to make the assignment of egress IPs simpler by defining the following sequence of events for a user wishing to define an egress IP for its cluster.
+
+As to satisfy 1.1. we propose that cluster admins tag the nodes they wish to use as egress nodes by labelling them with `node-role.kubernetes.io/egress-assignable: ""`. In case an `EgressIP` object is created and no nodes are tagged as such, all nodes are considered for egress IP assignment. To rectify that: a cluster admin would have to; delete the previously created `EgressIP` object, label the nodes needed, and re-create the previous object. 
+
+Adding a `PodSelector.MatchLabels` field (following the convention for such a field for Kubernetes / OpenShift) to the `EgressIP.Spec` as to allow for the added new functionality.
+
+As to satisfy all goal/requirements we propose the following EgressIP object definition:
+
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: EgressIP
+metadata:
+  name: egressip
+  namespace: egressip-namespace
+spec:
+  egressIPs:
+    192.168.126.11
+    192.168.126.102
+  podSelector:
+    matchLabels:
+      name: my-pod
+status:
+  assignments:
+  - node: nodeX
+    egressIP: 192.168.126.11
+  - node: nodeY
+    egressIP: 192.168.126.102
+```
+
+In case `nodeX` or `nodeY` go down, the assigned IP is re-distributed on another node in the cluster that is labelled for egress IP assignment (or all, in case no node is). 
+
+This feature simplifies the user interaction with this feature by:
+
+- Allowing the cluster admin not to dwelve in the details of finding out the node's subnet information
```

Comment:
"delve"

but that's not really true; they still have to "find out" the nodes' subnets so they know what ranges they can assign IPs out of. They just don't need to re-specify it here.

Reply:
Thanks. Yeah, I was basing myself on the fact that they _could_ chose not to tag any node at all and just start creating egress IPs based on the `machineNetworkCIDR` or something like that, thus simplifying their life a bit. But I'll remove that now.